src/process/clearHidden.ts

```diff
@@ -0,0 +1,23 @@
+import unset from 'lodash/unset';
+import {
+    ProcessorScope,
+    ProcessorContext,
+    ProcessorInfo,
+    ProcessorFnSync
+} from "types";
+
+/**
+ * This processor function checks components for the `hidden` property and unsets corresponding data
+ */
+export const clearHiddenProcess: ProcessorFnSync<ProcessorScope> = (context) => {
+    const { component, data, path, value } = context;
+    if (component.hidden && value !== undefined && (!component.hasOwnProperty('clearOnHide') || component.clearOnHide)) {
+        unset(data, path);
```

Comment:
we need a scope for this processor, which will have paths of the components that were unset

Reply:
ah, good call, will add